hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -590,26 +589,31 @@ public boolean syncDataFromOM() {
             Thread.currentThread().interrupt();
           } catch (Exception e) {
             metrics.incrNumDeltaRequestsFailed();
+            reconTaskUpdater.setLastTaskRunStatus(-1);
             LOG.warn("Unable to get and apply delta updates from OM.",
                 e.getMessage());
             fullSnapshot = true;
+          } finally {
```

Comment:
I had put it in finally in case there is any error while executing `getAndApplyDeltaUpdatesFromOM(...)`
In such a case the catch method will set last task status as -1 and then we update in the finally.

Shall I then have a nested try-block in just that part to catch exceptions while applying updates?

Reply:
Also follow up question. In the current PR we are also considering the `consumeOmEvents()` as a part of the task.
So we are also updating the status in the DB in the `ReconTaskControllerImpl` class. Shall we remove all updates from there as well? As the task controller is called during consumption.